pallets/gear/src/lib.rs

```diff
@@ -934,12 +952,61 @@ pub mod pallet {
         pub fn reset(origin: OriginFor<T>) -> DispatchResult {
             ensure_root(origin)?;
             <Mailbox<T>>::remove_all(None);
+            GearProgramPallet::<T>::reset_storage();
             common::reset_storage();
 
             Self::deposit_event(Event::DatabaseWiped);
 
             Ok(())
         }
+
+        /// Resumes a previously paused program
+        ///
+        /// The origin must be Signed and the sender must have sufficient funds to
+        /// transfer value to the program.
+        ///
+        /// Parameters:
+        /// - `program_id`: id of the program to resume.
+        /// - `memory_pages`: program memory before it was paused.
+        /// - `value`: balance to be transferred to the program once it's been resumed.
+        ///
+        /// - `ProgramResumed(H256)` in the case of success.
+        #[frame_support::transactional]
+        #[pallet::weight(<T as Config>::WeightInfo::resume_program(memory_pages.values().map(|p| p.len() as u32).sum()))]
+        pub fn resume_program(
+            origin: OriginFor<T>,
+            program_id: H256,
+            memory_pages: BTreeMap<u32, Vec<u8>>,
+            value: BalanceOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            let account = ensure_signed(origin)?;
+
+            ensure!(!value.is_zero(), Error::<T>::ResumeProgramNotEnoughValue);
+
+            GearProgramPallet::<T>::resume_program(
+                program_id,
+                memory_pages,
+                <frame_system::Pallet<T>>::block_number().unique_saturated_into(),
+            )
+            .map_err(|e| match e {
```

Comment:
Why not just return the respective error from the `gear-program` pallet (provided the public functions there themselves return a properly defined `pallet::error`s that can be sent to an external caller)?
For instance, we don't map errors from the balances pallet - just return as is.